File: qasdk/src/main/java/co/astrnt/qasdk/dao/JobApiDao.java
package co.astrnt.qasdk.dao;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

/**
 * Created by deni rohimat on 06/04/18.
 */
public class JobApiDao extends RealmObject {

    @PrimaryKey
    private long id;
    private String title;
    private int requireCv;
    private String description;
    private String location;
    private String requirement;
    private String responsibility;
    private String documentType;
    private String type;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public boolean isRequireCv() {
        return requireCv != 0;
    }

    public void setRequireCv(int requireCv) {
        this.requireCv = requireCv;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getRequirement() {
        return requirement;
    }

    public void setRequirement(String requirement) {
        this.requirement = requirement;
    }

    public String getResponsibility() {
        return responsibility;
    }

    public void setResponsibility(String responsibility) {
        this.responsibility = responsibility;
    }

    public String getDocumentType() {
        return documentType;
    }

    public void setDocumentType(String documentType) {
        this.documentType = documentType;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }
}

File: qasdk/src/main/java/co/astrnt/qasdk/dao/JobApiDao.kt
package co.astrnt.qasdk.dao

import com.google.gson.annotations.SerializedName
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

open class JobApiDao : RealmObject() {
    @PrimaryKey
    var id: Long = 0
    var title: String? = null
    private var requireCv = 0
    var description: String? = null
    var location: String? = null
    var requirement: String? = null
    var responsibility: String? = null
    var documentType: String? = null
    var type: String? = null

    @SerializedName("recruitment_type")
    var recruitmentType: String? = null

    @SerializedName("job_url")
    var jobUrl: String? = null

    fun isRequireCv(): Boolean {
        return requireCv != 0
    }

    fun setRequireCv(requireCv: Int) {
        this.requireCv = requireCv
    }
}